static/app/views/insights/pages/useOverviewPageTrackAnalytics.ts

```diff
@@ -0,0 +1,44 @@
+import {useEffect} from 'react';
+
+import {trackAnalytics} from 'sentry/utils/analytics';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import useProjects from 'sentry/utils/useProjects';
+import {useDomainViewFilters} from 'sentry/views/insights/pages/useFilters';
+
+export function useOverviewPageTrackPageload() {
+  const organization = useOrganization();
+  const pageFilters = usePageFilters();
+  const projects = useProjects();
+  const {view} = useDomainViewFilters();
+
+  const allProjects = projects.initiallyLoaded ? projects.projects : [];
+  const selectedProjects = allProjects.filter(p =>
+    pageFilters.selection.projects.includes(parseInt(`${p.id}`, 10))
+  );
+
+  // Stringifying this is just to avoid missing dependencies in the useEffect,
+  // Any performance implications are negligible for such a small array
+  const selectedPlatformsString = JSON.stringify([
+    ...new Set(
+      selectedProjects.map(project => project.platform).filter(p => p !== undefined)
+    ),
+  ]);
+
+  useEffect(() => {
+    if (pageFilters.isReady && projects.initiallyLoaded) {
+      const selectedPlatforms = JSON.parse(selectedPlatformsString);
+      trackAnalytics(`insights.page_loads.overview`, {
+        organization,
+        platforms: selectedPlatforms,
+        domain: view,
+      });
+    }
+  }, [
+    organization,
+    pageFilters.isReady,
+    projects.initiallyLoaded,
+    selectedPlatformsString,
+    view,
+  ]);
```

Comment:
👀 I'm not up-to-date on analytics stuff, but is it possible to use the [pre-built page load analytics](https://develop.sentry.dev/development-infrastructure/analytics/#customization) we have? Looks like it allows changing the name and the parameters. Might be safer

Reply:
I didn't know this was a thing, i'll have to take a look 👍

Reply:
I played around with this a bit, I just couldn't manage to get it to work. I think it could be because we need this for changes in the project selector in addition to page loads? I can see we use a similar pattern to track if insight modules have data or not, so I think it should be fine to merge this?

Reply:
Suspicious, but okay!